pallets/asset-manager/src/benchmarking.rs

```diff
@@ -20,121 +20,120 @@ use crate::{Call, Config, Event, Pallet};
     }
 
     mint_asset {
-        let start = <T::AssetConfig as AssetConfig<T>>::StartNonNativeAssetId::get();
-        let end = start + 1000;
-        for i in start..end {
-            let location = <T::AssetConfig as AssetConfig<T>>::AssetLocation::from(MultiLocation::new(0, X1(Parachain(i))));
-            let metadata = <T::AssetConfig as AssetConfig<T>>::AssetRegistrarMetadata::default();
+        let mut next = <T::AssetConfig as AssetConfig<T>>::StartNonNativeAssetId::get();
+        for i in 0..1000 {
+            let location = T::Location::from(MultiLocation::new(0, X1(Parachain(i))));
+            let metadata = <T::AssetConfig as AssetConfig<T>>::AssetRegistryMetadata::testing_default();
             Pallet::<T>::register_asset(RawOrigin::Root.into(), location.clone(), metadata.clone())?;
+            next = next.checked_add(&One::one()).unwrap();
         }
         let beneficiary: T::AccountId = whitelisted_caller();
         let amount = 100;
         // does not really matter what we register, as long as it is different than the previous
-        let location = <T::AssetConfig as AssetConfig<T>>::AssetLocation::default();
-        let metadata = <T::AssetConfig as AssetConfig<T>>::AssetRegistrarMetadata::default();
+        let location = T::Location::default();
+        let metadata = <T::AssetConfig as AssetConfig<T>>::AssetRegistryMetadata::testing_default();
         Pallet::<T>::register_asset(RawOrigin::Root.into(), location, metadata)?;
-    }: _(RawOrigin::Root, end, beneficiary.clone(), amount)
+    }: _(RawOrigin::Root, next, beneficiary.clone(), <T as Config>::Balance::from(amount) )
     verify {
-        assert_last_event::<T>(Event::AssetMinted { asset_id: end, beneficiary, amount }.into());
+        assert_last_event::<T>(Event::AssetMinted { asset_id: next, beneficiary, amount: <T as Config>::Balance::from(amount) }.into());
     }
 
     set_min_xcm_fee {
-        let start = <T::AssetConfig as AssetConfig<T>>::StartNonNativeAssetId::get();
-        let end = start + 1000;
-        for i in start..end {
+        let mut next = <T::AssetConfig as AssetConfig<T>>::StartNonNativeAssetId::get();
+        for i in 0..1000 {
 
             let location: MultiLocation = MultiLocation::new(0, X1(Parachain(i)));
-            let location = <T::AssetConfig as AssetConfig<T>>::AssetLocation::from(location.clone());
-            let metadata = <T::AssetConfig as AssetConfig<T>>::AssetRegistrarMetadata::default();
+            let location = T::Location::from(location.clone());
+            let metadata = <T::AssetConfig as AssetConfig<T>>::AssetRegistryMetadata::testing_default();
 
             Pallet::<T>::register_asset(RawOrigin::Root.into(), location.clone(), metadata.clone())?;
-            Pallet::<T>::set_units_per_second(RawOrigin::Root.into(), i, 0)?;
+            Pallet::<T>::set_units_per_second(RawOrigin::Root.into(), next, 0)?;
```

Comment:
what does this edit `i` -> `next` change?

Reply:
I think i was having some issues with the type checker.
I can just convert `i` an asset-id here